### PR TITLE
Create add-new-store-in-multi-db-setup.md

### PR DESCRIPTION
**Pull request by feversocial**
Created a "Adding new store in multi-db setup" howto guide

## PR Description
Adding and deploying a new store in multi-db store setup requires additional steps and preparation. This guideline contains all the best practices you need to know.

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+## Initial planning and best practices
+
+### Clear Roadmap
+It is good to have overall plan, detailing all stores that will be added in the future. This can impact not only database structure and configurations, but overall decisions on how to approach the rollout, making sure it is cost-efficient over time, on all ends.
+
+### Backup strategy
+Always have a backup plan ready in case of issues during the deployment. This includes not only database backups but also considerations on all points you will find below, including the business side.
+
+### Environment Preparation
+Prepare the development, staging, and production environments for a new store rollout. Make sure you don’t have additional functionality to be released on top or parallel development. This involves ensuring that teams are prepared and stakeholders are aware of the procedure.
```
Comment (Jackson88) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:22]: > Prepare the development, staging, and production environments for a new store rollout
Huge part of our customers have only staging and production by contract, I think it worth to mentioned somehow

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+last_updated: Dec 02, 2024
+template: howto-guide-template
+originalLink: 
+originalArticleId: 
+redirect_from:
+---
+
+Setting up a new store in an existing multi-DB environment requires a well-prepared plan, to avoid disrupting the data and operations of existing stores. This guide provides a detailed procedure to release a new store within a region that already hosts other stores, ensuring a non-destructive deployment.
+
+## Initial planning and best practices
```
Comment (Jackson88) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:13]: Points look very valid for all types of multi store setups

Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:13]: Agree, I know our existing documentation is incomplete - I tried to lookup multiple "multi store" articles we have - to link articles together, and it looks like a mess now.. but step by step we will improve it!

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+Follow [this guideline](/docs/scos/dev/technical-enhancement-integration-guides/integrate-multi-database-logic.html#define-databases) as a generic technical guideline, which is a requirement from the cloud team, to successfully deploy your new store in a multi-db setup.
+
+### Local Setup
+#### New store configuration
+
+* Define a new database and the store in the deploy file, following [that guide](/docs/ca/dev/multi-store-setups/add-and-remove-databases-of-stores.html#remove-the-configuration-of-the-database). As a result you should have:
+  * new database in regions.<region_name>.services.databases 
+  * new store in regions.<region_name>.stores 
+  * new domains in groups.<region_name>.applications  
+* Adjust stores.php with the configurations, relevant for your new store, following generic technical guideline.
```
Comment (profuel) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:59]: No Dynamic Multistore precondition is missing.

I have a string feeling that we're moving towards it, and thus touching this file is a bad practice.

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+* Define a new database and the store in the deploy file, following [that guide](/docs/ca/dev/multi-store-setups/add-and-remove-databases-of-stores.html#remove-the-configuration-of-the-database). As a result you should have:
+  * new database in regions.<region_name>.services.databases 
+  * new store in regions.<region_name>.stores 
+  * new domains in groups.<region_name>.applications  
+* Adjust stores.php with the configurations, relevant for your new store, following generic technical guideline. 
+* Prepare data import configurations and data files, specific to the new store. 
+* Adjust the local environment setup as needed, including configurations and environment variables. Examples: frontend router configuration, code bucket configuration, creating new backoffice users. 
+* Document all the steps you have done, to make sure they are repeatable in the future.
+
+#### Running Initial Setup
```
Comment (profuel) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:64]:
```suggestion
#### Running Initial Setup Locally
```

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+* Boot the new configurations using the appropriate setup tool.
```
Comment (profuel) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:66]: I tried, and it doesn't work locally. 
```
docker/sdk boot deploy.yml
docker/sdk run
```
results in the error that environment variables for the new store ur were not found.
```
/home/spryker/bin/cli.sh: line 17: /home/spryker/env/ur.env: No such file or directory
```

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+  * new domains in groups.<region_name>.applications  
+* Adjust stores.php with the configurations, relevant for your new store, following generic technical guideline. 
+* Prepare data import configurations and data files, specific to the new store. 
+* Adjust the local environment setup as needed, including configurations and environment variables. Examples: frontend router configuration, code bucket configuration, creating new backoffice users. 
+* Document all the steps you have done, to make sure they are repeatable in the future.
+
+#### Running Initial Setup
+
+* Boot the new configurations using the appropriate setup tool. 
+* Execute commands to initialize the database and services for the new store.
```
Comment (profuel) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:67]: which exactly? you never asked to prepare them.

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+It is convenient to create additional deployment install recipes (located under config/install folder) to setup a new and delete an existing stores, for testing purposes. Below is an example of such setup that proves to be working well on prcatice. We took the existing folder structure, and EU folder as a base, but you can introduce your structure:
+
+1. config/install/EU/setup-store.yml - contains everything needed to do a minimal setup of a new store(s):
+```
+env:
+  NEW_RELIC_ENABLED: 0
+command-timeout: 7200
+stores:
+  - { STORES_GO_HERE }
```
Comment (profuel) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:80]: if I'm not mistaken, they have to be placed each on a separate line

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+title: "Adding new store in multi-db setup"
+description: Adding and deploying a new store in multi-db store setup requires additional steps and preparation. This guideline contains all the best practices you need to know.
+last_updated: Dec 02, 2024
+template: howto-guide-template
+originalLink: 
+originalArticleId: 
+redirect_from:
+---
+
+Setting up a new store in an existing multi-DB environment requires a well-prepared plan, to avoid disrupting the data and operations of existing stores. This guide provides a detailed procedure to release a new store within a region that already hosts other stores, ensuring a non-destructive deployment.
```
Comment (andriitserkovnyi) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:11]: `non-destructive deployment` sounds like something related to `destructive deployment`. If it's not related, rephrase it

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+### Cloud environment and monitoring
+* Think of and adjust monitoring tools and APM (such as NewRelic, CloudWatch) to accommodate the new store. Check that all alerts and metrics are correctly configured to monitor the health and performance of the new store alongside existing ones.
+* Think of adjusting AWS Services such as S3, introducing buckets for the new store(s).
+
+### Front-end Considerations
+* Consider any other activities related to the above epics that might impact the deployment. For instance, front-end separation might be a significant task, requiring layout adjustments between different stores and possible adjustments on the API side.
+
+## Step-by-Step Procedure to release a new store(s)
+
+Follow [this guideline](/docs/scos/dev/technical-enhancement-integration-guides/integrate-multi-database-logic.html#define-databases) as a generic technical guideline, which is a requirement from the cloud team, to successfully deploy your new store in a multi-db setup.
```
Comment (andriitserkovnyi) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:50]: How is this document related to the linked document? What should I do with the linked document?

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 1 above)
Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md]: done, thank you!

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 5 above)
Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:66]: locally you would be running deploy.dev.yml;
+ from the error you're having it looks like you have "ur" instead of "us" ?

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 3 above)
Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:63]: You're right, added a warning in the very beginning

**Review by feversocial — COMMENTED** [review]
fixed according to comments

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 4 above)
Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md]: Fixed

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 6 above)
Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md]: Thanks, Changed sentences. THere's nothing special but running boot & up

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 7 above)
Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md]: yes, you're right - looks confusing. Fixed!

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 8 above)
Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md]: Please check, I changed the sentence

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 9 above)
Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md]: Thanks, Changed the sentence, please check

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 5 above)
Comment (profuel) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md]: I added new store into deploy.dev.yml and named it `ur`, and tried your recipe, and so it failed.

Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md]: @profuel I followed instructions on adding new store in mdb setup from here (as mentioned before): https://docs.spryker.com/docs/dg/dev/integrate-and-configure/integrate-multi-database-logic.html, 
And got 2 databases installed: 
Screenshot 2024-12-03 at 15 35 35

My docker/sdk up was not able to finish successfully since I have it broken from the beginning, but new DB creation works fine. 
To do the whole complete successful setup from scratch (with demo data etc), it would require me quite some time right now. What can we do here?

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+  * new store in regions.<region_name>.stores 
+  * new domains in groups.<region_name>.applications  
+* Adjust stores.php with the configurations, relevant for your new store, following generic technical guideline. 
+* Prepare data import configurations and data files, specific to the new store. 
+* Adjust the local environment setup as needed, including configurations and environment variables. Examples: frontend router configuration, code bucket configuration, creating new backoffice users. 
+* Document all the steps you have done, to make sure they are repeatable in the future.
+
+#### Running initial setup locally
+
+* Bootstrap your updated configuration and run your environment as usual, using {docker/sdk}(/docs/dg/dev/set-up-spryker-locally/install-spryker/install/install-in-development-mode-on-macos-and-linux.html#configure-and-start-the-instance)
```
Comment (profuel) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:70]: I still find this confusing.
I run my environment using `docker/sdk run`.
According to your comment, I have to use `docker/sdk up` in case of new DBs added.
Maybe it's worth being more specific here?

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+### Staging Setup
+#### Environment Configuration
+
+* Update the staging environment’s configuration to include the new store.
+* For database to be initialised, you will need to run a destructive deployment for your new store. To assure existing stores are not affected, you need to specify only new store code(s) in your deployment yml file (image.environment section), in `SPRYKER_HOOK_DESTRUCTIVE_INSTALL`. Example, for new PL and AT stores to be introduced:
+`SPRYKER_HOOK_DESTRUCTIVE_INSTALL: "vendor/bin/install PL,AT -r EU/destructive --no-ansi -vvv"`
+You can also use your custom recipe following the examples above (see “Setting up additional deployment recipes “)
+
+#### Support Requests
+* Open a support request to apply the new configuration to the staging environment. Attach all necessary files and provide detailed deployment instructions. You can also have the necessary configuration in a specific branch of your repository
```
Comment (profuel) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:134]: Let's list the files to attach to the request.
I'm also not sure which deployment instructions you mean, since Spryker own them, customers cannot change them.
```suggestion
* Open a support request to apply the new configuration to the environment. Attach deploy file and explain shortly expected changes, i.e. new DB should be created. In case you have the necessary configuration in a specific branch of your repository, provide a reference to it in the ticket, making sure support team has access to your code base.
```

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md (hunk 10 above)
Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:70]: every time you update your deploy file, you have to boot it, and after the bootstrap, you need to do docker/sdk up - as only that command goes through the install and runs different install sections (see https://docs.spryker.com/docs/dg/dev/sdks/the-docker-sdk/docker-sdk-quick-start-guide.html#setting-up-a-production-like-environment) 
docker/sdk start (I can't find in public documentation) only runs `docker-compose up -d` and restarts frontend and gateway (from here: https://docs.google.com/presentation/d/1whIbH8kdaOtTSzpWzUvUgeLoOMGZRL8b6sFdqHi8_ig/edit#slide=id.ge13b9c44a4_0_0)

I will mention boot & up, but let's not expand it even more, otherwise it will turn into docker/sdk manual

Comment (feversocial) [reply docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md]: Done, please have a look 👍

File: docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+* Bootstrap your updated configuration and run your environment as usual:
+  ```bash
+  docker/sdk boot deploy.dev.yml
+  docker/sdk up
+  ```
+* Verify that the new store’s database is correctly initialized and filled up with the demo data.
+
+#### Setting up additional deployment recipes
+
+It is convenient to create additional deployment install recipes (located under config/install folder) to setup a new and delete an existing stores, for testing purposes. Below is an example of such setup that proves to be working well on prcatice. We took the existing folder structure, and EU folder as a base, but you can introduce your structure:
```
Comment (andriitserkovnyi) [inline docs/dg/dev/internationalization-and-multi-store/add-new-store-in-multi-db-setup.md:79]: Existing where?
